client/logmon/logmon.go

```diff
@@ -0,0 +1,215 @@
+package logmon
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/driver/logging"
+	"github.com/hashicorp/nomad/client/lib/fifo"
+)
+
+const (
+	// processOutputCloseTolerance is the length of time we will wait for the
+	// launched process to close its stdout/stderr before we force close it. If
+	// data is written after this tolerance, we will not capture it.
+	processOutputCloseTolerance = 2 * time.Second
+)
+
+type LogConfig struct {
+	// LogDir is the host path where logs are to be written to
+	LogDir string
+
+	// StdoutLogFile is the path relative to LogDir for stdout logging
+	StdoutLogFile string
+
+	// StderrLogFile is the path relative to LogDir for stderr logging
+	StderrLogFile string
+
+	// StdoutFifo is the path on the host to the stdout pipe
+	StdoutFifo string
+
+	// StderrFifo is the path on the host to the stderr pipe
+	StderrFifo string
+
+	// MaxFiles is the max rotated files allowed
+	MaxFiles int
+
+	// MaxFileSizeMB is the max log file size in MB allowed before rotation occures
+	MaxFileSizeMB int
+
+	// UID is id for the desired user to write log files as
+	UID int
+
+	// GID is id for the desried group to write log files as
+	GID int
```

Comment:
Could we just run the logmon process as this user and group? We wouldn't have to plumb these through or chown files on creation.